### ISSUE TITLE
Plug in seems to be active, accessories visible in Homebridge, but not responding in Homekit

### ISSUE THREAD

**Issue by macguyincali** [node-alarm-dot-com/homebridge-node-alarm-dot-com#107]
I've had this plugin for about a month or two and it has worked swimmingly. For some reason, last night the panel wouldn't respond via homekit and now none of the sensors appear in Homekit. The panel shows "not responding." When I look at Homebridge, the accessories are all there and the sensors appear active and reflecting their correct states.  I've restarted the child node and also restarted Homebridge, but I'm still having the problem. Not sure if this is a bug request or just a dumb user messed something up. My apologies if the latter.

**Comment by chase9** [comment]
Could you post your logs with any sensitive information scrubbed?Chase LauPlease excuse any grammatical errors as this was sent from my iPhone.On Feb 2, 2023, at 6:31 PM, macguyincali ***@***.***> wrote:﻿
I've had this plugin for about a month or two and it has worked swimmingly. For some reason, last night the panel wouldn't respond via homekit and now none of the sensors appear in Homekit. The panel shows "not responding." When I look at Homebridge, the accessories are all there and the sensors appear active and reflecting their correct states.  I've restarted the child node and also restarted Homebridge, but I'm still having the problem. Not sure if this is a bug request or just a dumb user messed something up. My apologies if the latter.

**Comment by MAscooby** [comment]
That sounds a lot like what I've been seeing, but the behavior extends to all of my Homebridge plugins when this plugin is enabled. Devices show up in the Home app, but if I try to use a device in the Home app the device indicator toggles back to its prior state and becomes unusable. I had assumed it was something to do with HomeKit changes until someone on reddit figured out that rolling this plug-in back to the prior version resolved the issue.

**Comment by etonnis** [comment]
[Reddit link](https://www.reddit.com/r/homebridge/comments/10rph6r/all_accessories_showing_no_response_but/j70zjpb/?utm_source=share&utm_medium=ios_app&utm_name=iossmf&context=3) 

I was the Reddit poster (link above). My system appears to be stable after 24 hrs on v1.8.0 - with no recurring loss of communication with HomeKit. Let me know if there’s any info I can share that would be helpful in tracking down the root cause of the issue.

**Comment by chase9** [comment]
Hi guys, I want to get this fixed ASAP but I need logs to diagnose the issues. You can post them here or DM me on Discord if you prefer.

**Comment by etonnis** [comment]
@chase9 My prior logs didn't capture the issue when it emerged. I'm trying to reproduce now (with log in debug mode) and will let you know if I can capture it. In both prior cases, it took between 4-12 hours after upgrade to v 1.9.0 for the issue to emerge. I've since moved the plugin to its own child bridge so not sure if it will reproduce in this environment...

**Comment by etonnis** [comment]
@chase9  I have finally had a recurrence of the HomeKit dropout of the v1.9.0 plug-in in the child bridge...trying to extract logs for past few hours and scrub them...

**Comment by kingkaden123** [comment]
Having this same issue. Tried reinstalling but I'm at the point where all sensors are showing as open/activated. The panel is showing as intended though. 

Not seeing any logs about anything but the glass sensors (Which are the only ones not showing in HomeKit) 
[2/5/2023, 12:01:06 AM] [Security System] Warning: Sensor Family Room Glass Break has unknown state 0 (SensorNumber)

Hope this helps, will answer any questions if necessary.

**Comment by jclarkenz** [comment]
1.8.0 causes my entire Homebridge device setup to be Not Responding in HomeKit. Reverting to 1.7.0 fixes the issue.

**Comment by etonnis** [comment]
I've not had any issues with 1.8.0 - only with the upgrade to 1.9.0. With the plug-in operating in a dedicated child bridge, only the Alarm.com security system is now dropping out (after 4-12 hrs of operating ok). 

@chase9 My log seems to be truncated to a max 2 hr length - and I keep missing capture of the issue. If you contact me directly, you might be able to help me with the right HB settings to capture the right log at the right time to help debug the issue. I can be reached at: varnish_decoy.0d [at] icloud.com.

**Comment by etonnis** [comment]
I've had one recurrence of 1.9.0 dropping out - but this one coincided with the v1.6.3 update to my HomePod and AppleTV software this afternoon - so not sure how relevant. 

The logs (even in debug mode) all seem show nothing unusual - the periodic login to Alarm.com keeps being shown, but none of the sensors respond anymore past a certain point. Everything is still functional in Homebridge, just not in HomeKit. 

One other observation - not sure if it's relevant or not - around the time of each loss of HomeKit functionality, my Ubiquiti DMP-SE has informed me of an intrusion attempt from Indonesia. (see attached) It seems more than coincidental by now, but I don't know what to make of it. ![image](https://user-images.githubusercontent.com/124382866/217148372-6d52ae35-c568-48cb-93e8-db07fcc79b84.png)

**Comment by chase9** [comment]
@kingkaden123 Do you see duplicates coming from the plugin? I'm wondering if the sensor upgrade created new accessories and the old ones are stuck in their state.

**Comment by kingkaden123** [comment]
@chase9 Sorry for the late reply. Duplicates are not showing from the plugin. All I see are all the sensors in the open states. I can screenshot from HomeKit if you need me to.

**Comment by bedobtz** [comment]
Same exact issue here as well since iOS, HomePod and tvOS updates. I tried reverting back down to 1.80 version of plugin but didn’t help

**Comment by MAscooby** [comment]
> tvOS

If you haven't already you might need to delete and reset the bridge, and then re-add it under 1.8

**Comment by kaganae** [comment]
FWIW, I had the same issue. I have been using the plug-in for several years and this just started happening after the 1.9 update. I removed the cached accessories (I noticed that I had 2 cached Panel accessories, oddly) and restarted. It seems to be working fine after that.

**Comment by majorgearhead** [comment]
> 1.9.0 causes my entire Homebridge device setup to be Not Responding in HomeKit. Reverting to 1.8.0 fixes the issue.

It did for me as well but I found a fix.  I changed the mDNS responder on Homebridge to Ciao, removed the accessory cache, restarted homebridge, and restarted my UniFi network devices.

After that all of the duplicates were gone and all of the "Non Responding" went away.  All devices are able to respond.

I chalk it up to the mDNS responder and in my case a more recent UnifiOS update.  Maybe this will work for you?
